gobblin-core/src/main/java/gobblin/source/extractor/extract/google/webmaster/GoogleWebmasterClientImpl.java

```diff
@@ -0,0 +1,98 @@
+package gobblin.source.extractor.extract.google.webmaster;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.repackaged.com.google.common.base.Preconditions;
+import com.google.api.services.webmasters.Webmasters;
+import com.google.api.services.webmasters.WebmastersScopes;
+import com.google.api.services.webmasters.model.ApiDataRow;
+import com.google.api.services.webmasters.model.ApiDimensionFilter;
+import com.google.api.services.webmasters.model.ApiDimensionFilterGroup;
+import com.google.api.services.webmasters.model.SearchAnalyticsQueryRequest;
+import com.google.api.services.webmasters.model.SearchAnalyticsQueryResponse;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class GoogleWebmasterClientImpl implements GoogleWebmasterClient {
+  private final static Logger LOG = LoggerFactory.getLogger(GoogleWebmasterClientImpl.class);
+
+  private final Webmasters.Searchanalytics _analytics;
+
+  public GoogleWebmasterClientImpl(String credentialFile, String appName, String scope) throws IOException {
+    Preconditions.checkArgument(
+        Objects.equals(WebmastersScopes.WEBMASTERS_READONLY, scope) || Objects.equals(WebmastersScopes.WEBMASTERS,
+            scope), "The scope for WebMaster must either be WEBMASTERS_READONLY or WEBMASTERS");
+
+    GoogleCredential credential =
+        GoogleCredential.fromStream(new FileInputStream(credentialFile)).createScoped(Collections.singletonList(scope));
+    //GoogleNetHttpTransport.newTrustedTransport(),
+    //JacksonFactory.getDefaultInstance(),
+    Webmasters service =
+        new Webmasters.Builder(new NetHttpTransport(), new JacksonFactory(), credential).setApplicationName(appName)
+            .build();
+    _analytics = service.searchanalytics();
+  }
+
+  @Override
+  public List<String> getPages(String siteProperty, String date, GoogleWebmasterFilter.Country country, int rowLimit,
```

Comment:
I think it may be an easy and worthwhile change to return a Callable<> which does the actual call. Thus we can later parallelize the calls in the fetcher getAllPages().